### PR TITLE
fix(nuxt): check vnode element instead of hydration state for islands

### LINKED ISSUE #26434: Server components cause hydration error when used inside resolved component
### Environment

------------------------------
- Operating System: Linux
- Node Version:     v18.18.0
- Nuxt Version:     3.11.1
- CLI Version:      3.11.1
- Nitro Version:    2.9.4
- Package Manager:  npm@10.2.3
- Builder:          -
- User Config:      devtools
- Runtime Modules:  -
- Build Modules:    -
------------------------------

### Reproduction

https://stackblitz.com/edit/github-v2dhfq

### Describe the bug

Since Nuxt version 3.10 server components cause an hydration mismatch and don't render when used inside resolved components.

### Additional context

_No response_

### Logs

```shell-script
[Vue warn]: Hydration node mismatch:
- rendered on server: <!--[--> (start of fragment) 
- expected on client: div 
  at <NuxtIsland name="Icon" lazy=false props= Proxy(Object)  ... > 
  at <Icon> 
  at <Container.global> 
  at <AsyncComponentWrapper> 
  at <App key=3 > 
  at <NuxtRoot>
```

### PR DESCRIPTION
**Pull request by huang-julien**
### 🔗 Linked issue
fix #26434

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Hi :wave: this PR prefer using `vnode.el` instead of `nuxtApp.isHydrating`

Not sure why we do have this issue with `resolveComponent`.

This PR however fix the case where nuxtIsland is called within an async component

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have added tests (if possible).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nuxt/src/app/components/nuxt-island.ts [hunk 1]
```diff
@@ -109,14 +109,14 @@ export default defineComponent({
...
+    if (instance.vnode.el) {
       payloads.slots = toRaw(nuxtApp.payload.data[`${props.name}_${hashId.value}`])?.slots ?? {}
       payloads.components = toRaw(nuxtApp.payload.data[`${props.name}_${hashId.value}`])?.components ?? {}
     }
 
     const ssrHTML = ref<string>('')
 
-    if (import.meta.client && nuxtApp.isHydrating) {
+    if (import.meta.client && instance.vnode.el) {
       ssrHTML.value = getFragmentHTML(instance.vnode?.el ?? null, true)?.join('') || ''
```
Comment (danielroe) [inline packages/nuxt/src/app/components/nuxt-island.ts:120]:
```suggestion
    if (import.meta.client && instance.vnode?.el) {
      ssrHTML.value = getFragmentHTML(instance.vnode.el, true)?.join('') || ''
```

File: packages/nuxt/src/app/components/nuxt-island.ts [hunk 2]
```diff
@@ -250,7 +250,7 @@ export default defineComponent({
           // this is used to force trigger Teleport when vue makes the diff between old and new node
           const isKeyOdd = teleportKey.value === 0 || !!(teleportKey.value && !(teleportKey.value % 2))
 
-          if (uid.value && html.value && (import.meta.server || props.lazy ? canTeleport : mounted.value || nuxtApp.isHydrating)) {
+          if (uid.value && html.value && (import.meta.server || props.lazy ? canTeleport : mounted.value || instance.vnode.el)) {
```
Comment (danielroe) [inline packages/nuxt/src/app/components/nuxt-island.ts:253]:
```suggestion
          if (uid.value && html.value && (import.meta.server || props.lazy ? canTeleport : mounted.value || instance.vnode?.el)) {
```

**Comment by danielroe** [comment]
Might be the same issue as https://github.com/vuejs/core/issues/6638.

**Comment by huang-julien** [comment]
@danielroe it's probably the same issue. Is it better to let vue/core fix it ?

**Comment by danielroe** [comment]
I think this is a good fix in the meantime. But fixing upstream would be 👌

**Review by danielroe — APPROVED** [review]

**Pull request merged**